### PR TITLE
Domain layer finished

### PR DESCRIPTION
**Pull request by albertoag128**
## DESCRIPTION
This pull request is about showing how the domain layer has been implemented. In first instance we have our domain models, some use cases and also a repository which will be providing methods for the use cases.

## HOW IT WAS IMPLEMENTED
We can find here 4 files:
-->Models: this file will contain our domain model of movies and we could also add here some logic business.
-->Use cases: we have two of them; the one that has been created in order to get the movies feed and the second one focused on find just one movie by its id. Both of this files contains its own data class, because in the end we just want to get the information that we will be showing on the UI through the use cases.
-->MoviesRepository: this will be our interface to define what are the methods that we will be using in our use cases.

## KEYWORDS
Domain, Models, Repository, Data Classes, Use Cases, Interface, Separation of concerns, Clean arquitecture

## SCREENSHOTS
Domain structure
imagen

### CODE REVIEW

File: app/src/main/java/com/example/moviesapp/domain/GetMoviesFeedUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+                it.title,
+                it.year,
+                it.rating
+            )
+        }
+        return moviesList
+    }
+}
+
+data class MoviesFeed(val poster:String, val title:String, val year:String, val rating:String)
```
Comment (josem-perram) [inline app/src/main/java/com/example/moviesapp/domain/GetMoviesFeedUseCase.kt:19]: MoviesFeed is the response of use case so it should be inside the use case.

File: app/src/main/java/com/example/moviesapp/domain/GetSingleMovieUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+            movie.title,
+            movie.plot,
+            movie.genre,
+            movie.rating,
+            movie.year
+        )
+    }
+}
+
+data class MovieDetail(val poster:String, val title:String, val plot:String, val genre:String, val rating: String, val year:String)
```
Comment (josem-perram) [inline app/src/main/java/com/example/moviesapp/domain/GetSingleMovieUseCase.kt:18]: the same, MovieDetail should be inside the use case ;)

**Review by josem-perram — CHANGES_REQUESTED** [review]
The PR is ok, just some comments about where putting the model of use case.

**Comment by albertoag128** [comment]
The changes requested have been applied to the project moving the data classes of the use cases inside the use cases. I will close this pull request with this comment, and after that, I will delete domain branch and merge it with master.

**Pull request merged**